### PR TITLE
sql: add missing dynamic privileges and update optm lock version

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add missing dynamic privileges and update the version that supports optimistic locking to v3.0.8.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/13050, https://github.com/pingcap/docs-cn/pull/12492
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: privilege-management.md [hunk 1]
```diff
@@ -192,9 +192,17 @@ Dynamic privileges include:
...
 * `ROLE_ADMIN`
 * `CONNECTION_ADMIN`
-* `SYSTEM_VARIABLES_ADMIN`
+* `PLACEMENT_ADMIN` allows privilege owners to create, modify, and remove placement policies.
+* `DASHBOARD_CLIENT` allows privilege owners to log in to the TiDB-Dashboard.
+* `RESTRICTED_TABLES_ADMIN` allows privilege owners to see system tables when SEM is enabled.
+* `RESTRICTED_STATUS_ADMIN` allows privilege owners to see all status vars when SEM is enabled.
+* `RESTRICTED_VARIABLES_ADMIN` allows privilege owners to see all variables when SEM is enabled.
+* `RESTRICTED_USER_ADMIN` prohibits privilege owners to have their access revoked by SUPER users.
+* `RESTRICTED_CONNECTION_ADMIN` prohibits privilege owners to have connections killed by the `PROCESS` or `CONNECTION_ADMIN` privilege.
```
Comment (djshow832) [inline privilege-management.md:205]: More details in the RFC: 
Killing connections that belong to your own user is always permitted. Killing connections that belong to other users requires `CONNECTION_ADMIN` or `SUPER`; but there is an exception for `RESTRICTED_USER_ADMIN` users. **To kill these connections also requires `RESTRICTED_CONNECTION_ADMIN`**. This affects `KILL`, `KILL TIDB` commands.

File: privilege-management.md [hunk 2]
```diff
@@ -192,9 +192,17 @@ Dynamic privileges include:
...
+* `SYSTEM_VARIABLES_ADMIN`
 * `ROLE_ADMIN`
 * `CONNECTION_ADMIN`
-* `SYSTEM_VARIABLES_ADMIN`
+* `PLACEMENT_ADMIN` allows privilege owners to create, modify, and remove placement policies.
+* `DASHBOARD_CLIENT` allows privilege owners to log in to the TiDB-Dashboard.
+* `RESTRICTED_TABLES_ADMIN` allows privilege owners to see system tables when SEM is enabled.
+* `RESTRICTED_STATUS_ADMIN` allows privilege owners to see all status vars when SEM is enabled.
+* `RESTRICTED_VARIABLES_ADMIN` allows privilege owners to see all variables when SEM is enabled.
+* `RESTRICTED_USER_ADMIN` prohibits privilege owners to have their access revoked by SUPER users.
```
Comment (djshow832) [inline privilege-management.md:204]:
```suggestion
* `RESTRICTED_USER_ADMIN` prohibits privilege owners to have their access revoked by SUPER users when SEM is enabled.
```

File: privilege-management.md (hunk 1 above)
Comment (shichun-0415) [inline privilege-management.md:205]:
```suggestion
* `RESTRICTED_CONNECTION_ADMIN` allows privilege owners to kill connections of `RESTRICTED_USER_ADMIN` users. This affects `KILL` and `KILL TIDB` commands.
```

Comment (shichun-0415) [reply privilege-management.md:205]: @djshow832 PTAL

Comment (djshow832) [reply privilege-management.md:205]: LGTM

**Review by djshow832 — APPROVED** [review]

File: privilege-management.md [hunk 3]
```diff
@@ -192,9 +192,17 @@ Dynamic privileges include:
...
 * `ROLE_ADMIN`
 * `CONNECTION_ADMIN`
-* `SYSTEM_VARIABLES_ADMIN`
+* `PLACEMENT_ADMIN` allows privilege owners to create, modify, and remove placement policies.
+* `DASHBOARD_CLIENT` allows privilege owners to log in to TiDB Dashboard.
+* `RESTRICTED_TABLES_ADMIN` allows privilege owners to view system tables when SEM is enabled.
+* `RESTRICTED_STATUS_ADMIN` allows privilege owners to view all status variables in [`SHOW [GLOBAL|SESSION] STATUS`](/sql-statements/sql-statement-show-status.md) when SEM is enabled.
+* `RESTRICTED_VARIABLES_ADMIN` allows privilege owners to view all system variables when SEM is enabled.
+* `RESTRICTED_USER_ADMIN` prohibits privilege owners to have their access revoked by SUPER users when SEM is enabled.
+* `RESTRICTED_CONNECTION_ADMIN` allows privilege owners to kill connections of `RESTRICTED_USER_ADMIN` users. This privilege affects `KILL` and `KILL TIDB` commands.
```
Comment (Oreoxmt) [inline privilege-management.md:205]:
```suggestion
* `RESTRICTED_CONNECTION_ADMIN` allows privilege owners to kill connections of `RESTRICTED_USER_ADMIN` users. This privilege affects `KILL` and `KILL TIDB` statements.
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**